src/components/ProgressIndicator/ProgressIndicator.jsx

```diff
@@ -1,141 +1,362 @@
+      </span>
+    );
+  };
+
+  const StepLine = () => {
+    const classes = classnames({
+      [`line`]: !complete && !subStep,
+      [`line-sub`]: !complete && subStep,
+      [`line-complete`]: complete && !subStep,
+      [`line-sub-complete`]: complete && subStep,
+    });
+
+    return !lastItem ? <div className={classes} /> : null;
+  };
+
+  const StepLabel = () => {
+    const classes = classnames({
+      [`label`]: mainStep || subStep,
+      [`hidden`]: !showLabel && !current,
+    });
+
+    return (
+      <p className={classes} value={description}>
+        {label}
+      </p>
+    );
+  };
+
+  const StepSecondaryLabel = () => {
+    const classes = classnames({
+      [`label-optional`]: mainStep || subStep,
+      [`hidden`]: !showLabel && !current,
+    });
+
+    return secondaryLabel !== null && secondaryLabel !== undefined ? (
+      <p className={classes}>{secondaryLabel}</p>
+    ) : null;
+  };
+
+  const StepButton = () => {
+    const classes = classnames({
+      [`step-button`]: true,
+      [`main-step`]: mainStep,
+      [`sub-step`]: subStep,
+      [`clickable`]: accessible,
+    });
+
+    // for testing purposes
+    const dataTestIdLabel = label.replace(/\s/g, '-').toLowerCase();
+    const type = mainStep ? 'main' : 'sub';
+
+    return (
+      <>
+        <button
+          className={classes}
+          type="button"
+          aria-disabled={disabled}
+          disabled={disabled}
+          style={getStepWidth()}
+          onClick={accessible ? handleClick : null}
+          onKeyDown={accessible ? handleKeyDown : null}
+          data-testid={`step-button-${type}-${dataTestIdLabel}`}
+        >
+          <StepLine />
+          <StepIcon />
+          <div className="label-container">
+            <StepLabel />
+            <StepSecondaryLabel tabIndex="-1" />
+          </div>
+        </button>
+      </>
+    );
+  };
+
+  const classes = classnames({
+    [`step-current`]: current && !disabled,
+    [`step-complete`]: complete && !disabled && !invalid,
+    [`step-incomplete`]: incomplete && !current && !disabled && !invalid,
+    [`step-disabled`]: disabled,
+    [`step-invalid`]: invalid,
+  });
+
+  return (
+    <li className={classes}>
+      <StepButton tabIndex="0" />
+    </li>
+  );
+};
+
+ProgressStep.propTypes = {
+  id: IDPropTypes.isRequired,
+  label: PropTypes.string,
+  secondaryLabel: PropTypes.string,
+  description: PropTypes.string,
+  onChange: PropTypes.func,
+  disabled: PropTypes.bool,
+  showLabel: PropTypes.bool,
+  stepWidth: PropTypes.number,
+  vertical: PropTypes.bool,
+  invalid: PropTypes.bool,
+  stepNumber: PropTypes.number,
+  lastItem: PropTypes.bool,
+  isClickable: PropTypes.bool,
+  current: PropTypes.bool,
+  complete: PropTypes.bool,
+  incomplete: PropTypes.bool,
+  mainStep: PropTypes.bool,
+  subStep: PropTypes.bool,
+};
+
+ProgressStep.defaultProps = {
+  label: null,
+  secondaryLabel: null,
+  description: null,
+  onChange: null,
```

Comment:
we should use our default function so component does not blow up if no `onChange` is provided.

Reply:
Or nullcheck before calling the function https://github.com/IBM/carbon-addons-iot-react/issues/1401